### ISSUE TITLE
Nextcloud Beta 20170105: Auto upload preview doesn't shows content when Wifi enabled

### ISSUE THREAD

**Issue by OlivierS1** [nextcloud/android#511]
### Actual behaviour
-Tell us what happens
When accessing the Auto upload settings, if Wifi is enabled and connection to server is valid, the pictures will not be displayed. Instead, an icon used as a place holder will be shown.
Disabling Wifi while application is running will force Nextloud app in displaying the previews.
It seems the pictures will eventually render after a long while.

### Expected behaviour
-Tell us what should happen
Pictures should be rendered fairly quickly no matter the state of Wifi.
 
### Steps to reproduce
1. Start with WiFi enabled.
2. Access Auto upload menu.
3. Pictures should be populated with photos, but only icons are present.
3 A. If photo showing, close app and restart it following above steps.
4. To have icons replaced with actual pictures, disable Wifi.
5. Back in Auto Upload menu, photo should now be visible.

### Environment data
Android version:
6.0

Device model: 
LG-G3 LG-D852

Stock or customized system:
Stock

Nextcloud app version:
Beta 20170105

Nextcloud server version:
OC 9.1

#### Web server error log
```
Insert your webserver log here
```

#### Nextcloud log (data/nextcloud.log)
```
Insert your Nextcloud log here
```

**Comment by OlivierS1** [comment]
Updated to Nextcloud Beta 20170109:
"Auto upload" folder content is now being displayed properly when Wifi is enabled.
However, and it maybe just be the way it works or may require some tuning, but if I take a picture, while Nextcloud app is still in the background, the content will not update unless the "Auto upload" folder folder is closed and reopened.
It would be nice the contents be updated in realtime without having to close and reopen the "Auto upload" folder/menu.

**Comment by AndyScherzinger** [comment]
Hi @OlivierS1,
this seems weird since the auto upload view isn't related to the wifi connectivity. The app simply executes a Android Media API query and then displays loads the images, what might happen though is that something goes wrong while loading the image (that would be OS level) and thus the placeholder gets displayed. With the "switch to camera, take pictures, switch back to auto upload" issue, that is something where we can definitely improve the behavior though the question is then what would be the best solution.

* In order to detect a change we would have to execute the media query again and check if any "folder" has changed regarding the images actually displayed and if so, update the list (or folder) which cost some processing power (which yes, might be fine) or
* have  "pull to refresh" implemented which would reload the data (like mentioned above but just on-demand then)

**Comment by tobiasKaminsky** [comment]
When you see the default preview icon, please pull down to reload? Does this help to generate the thumbnails?
WLAN indeed should not be a trigger as the upload files are still on the device and can be generated without network connection.
I suspect this is just a race problem: the images may be generated but not be shown.

**Comment by OlivierS1** [comment]
Hi Tobias,
If you mean dragging the finger down the screen while in the "auto upload" page, then, no. I do not see the reload (circling overlay). 
However, the content may eventually appear.

**Comment by AndyScherzinger** [comment]
@tobiasKaminsky for the auto upload view we didn't implement the pull-to-refresh yet but I do have this on my agenda for the next (feature) release. Will open an issue for it in the next days.

**Comment by mario** [comment]
Can this be considered solved and closed?

**Comment by tobiasKaminsky** [comment]
I do not remember any fix regarding this issue?

**Comment by OlivierS1** [comment]
I have experienced some issues with the last two beta APK's. Not sure if it is application specific. But I can tell that "Auto upload" does seem to work on a new install of 20170209.APK. 

I still suspect there are latencies related to server communication causing delays in the thumbnail preview generation when accessing over the internet. 
Accessing the server locally over wifi doesn't exhibit any issues, just a somewhat marginal delay before the first thumbnail display its preview..

This past few weeks I was in the process of migrating my server to different drives configuration and as caching doesn't seem to work properly with my NC 11.0.1, it is possible the server was causing extra delays for the phone to get the data needed.

However, it is interesting to note that scrolling through the "Auto upload" grid view, the first thing done is displaying the default thumbnail icons which is going to be updated with the actual previews.
If the app is still processing other folders while the scroll is underway, then the thumbnail that had the previews are going to revert back to their default thumbnail icons.

**Comment by tobiasKaminsky** [comment]
I tried it with emulator, there you can set the network type and signal strength. I set it to "gsm" and "poor" and did a full account sync. While this I accessed "auto upload view" and all thumbnails were generated really fast.

**Comment by xXSTrikeXx** [comment]
Can be closed? V2.0 resolves this?

**Comment by OlivierS1** [comment]
It could be closed as it was based on Dev version.
